### PR TITLE
chore: Fix hundreds of typos in 200+ files.

### PR DESCRIPTION
**Pull request by Sajjon**
Fix hundreds of typos in 200+ files using [`typos` (Rust) cli tool](https://crates.io/crates/typos-cli) + 2 hours of manual work removing false positives.

**There might be some false positives left**.

I've not run any tests yet | which I ofc will do, I'm reading [dev guide about testing now](https://rustc-dev-guide.rust-lang.org/tests/running.html).

But I wanted to get feedback from Rust Community if you want me to break this PR up in many PRs? [I've recently done similar typos (fixing typos is a hobby of mine) in SwiftLang, where I split the work in 14 PRs](https://github.com/swiftlang/swift/issues/75022)- maybe something similar would be appriciated?

### CODE REVIEW

File: compiler/rustc_resolve/messages.ftl [hunk 1]
```diff
@@ -11,7 +11,7 @@ resolve_added_macro_use =
 resolve_ancestor_only =
     visibilities can only be restricted to ancestor modules
 
-resolve_anonymous_livetime_non_gat_report_error =
+resolve_anonymous_lifetime_non_gat_report_error =
```
Comment (workingjubilee) [inline compiler/rustc_resolve/messages.ftl:14]: This one will change diagnostics.

File: library/core/src/array/mod.rs [hunk 2]
```diff
@@ -37,7 +37,7 @@ pub use iter::IntoIter;
 ///
 /// # Example
 ///
-/// Creating muliple copies of a `String`:
+/// Creating multiple copies of a `String`:
```
Comment (workingjubilee) [inline library/core/src/array/mod.rs:40]: Note that despite my comments about altering compiler behavior, I would be happy to review and approve a PR which only touched the code in the library directory.

**Review by workingjubilee — COMMENTED** [review]
This is a "typo-fixing" PR that will probably have effects on the compiler's behavior. I found at least one case. This is why T-compiler passed this decision: https://github.com/rust-lang/rust/issues/58619

I will leave it to them to decide what to do with this, but the problem is that this change does not come with automation to verify we don't add more typos.

File: compiler/rustc_resolve/messages.ftl (hunk 1 above)
Comment (workingjubilee) [reply compiler/rustc_resolve/messages.ftl:14]: Oh, actually it just fails to compile, heh.

File: tests/ui/mir/issue-29227.rs [hunk 3]
```diff
@@ -47,12 +47,12 @@ macro_rules! match_char_class {
...
... '\u{A64E}', '\u{A650}', '\u{A652}', '\u{A654}', '\u{A656}', '\u{A658}', '\u{A65A}', '\u{A65C}', '\u{A65E}', '\u{A662}', '\u{A664}', '\u{A666}', '\u{A668}', '\u{A66A}', '\u{A66C}', '\u{A680}', '\u{A682}', '\u{A684}', '\u{A686}', '\u{A688}', '\u{A68A}', '\u{A68C}', '\u{A68E}', '\u{A690}', '\u{A692}', '\u{A694}', '\u{A696}', '\u{A722}', '\u{A724}', '\u{A726}', '\u{A728}', '\u{A72A}', '\u{A72C}', '\u{A72E}', '\u{A732}', '\u{A734}', '\u{A736}', '\u{A738}', '\u{A73A}', '\u{A73C}', '\u{A73E}', '\u{A740}', '\u{A742}', '\u{A744}', '\u{A746}', '\u{A748}', '\u{A74A}', '\u{A74C}', '\u{A74E}', '\u{A750}', '\u{A752}', '\u{A754}', '\u{A756}', '\u{A758}', '\u{A75A}', '\u{A75C}', '\u{A75E}', '\u{A760}', '\u{A762}', '\u{A764}', '\u{A766}', '\u{A768}', '\u{A76A}', '\u{A76C}', '\u{A76E}', '\u{A779}', '\u{A77B}', '\u{A77D}', '\u{A77E}', '\u{A780}', '\u{A782}', '\u{A784}', '\u{A786}', '\u{A78B}', '\u{FF21}', '\u{FF22}', '\u{FF23}', '\u{FF24}', '\u{FF25}', '\u{FF26}', '\u{FF27}', '\u{FF28}', '\u{FF29}', '\u{FF2A}', '\u{FF2B}', '\u{FF2C}', '\u{FF2D}', '\u{FF2E}', '\u{FF2F}', '\u{FF30}', '\u{FF31}', '\u{FF32}', '\u{FF33}', '\u{FF34}', '\u{FF35}', '\u{FF36}', '\u{FF37}', '\u{FF38}', '\u{FF39}', '\u{FF3A}')
```
Comment (saethlin) [inline tests/ui/mir/issue-29227.rs:50]: This and the other changes in this file are false positives

**Comment by saethlin** [comment]
> But I wanted to get feedback from Rust Community if you want me to break this PR up in many PRs?

Yes. Different parts of the directory tree are owned by different groups of reviewers; you can tell by the mass of labels that was applied to this PR that it touches many separate areas.

In addition to what Jubilee said, I'd be happy to review:
* a PR of only comment fixes in `compiler/` except `compiler/rustc_codegen_gcc`
* a PR of any fixes to `src/tools/miri`
* a PR of any fixes to `tests/`

**Comment by matthiaskrgr** [comment]
> fixing typos is a hobby of mine

have you considered writing a program that introduces subtle typos into a given document (switching letters or deleting random chracters etc)? This way you could fulfill you hobby of fixing typos as much as you would like without having to rely on other peoples time and review capacity. :slightly_smiling_face:

**Comment by saethlin** [comment]
@matthiaskrgr Nobody is asking for your time. Please try to be kind.

**Review by compiler-errors — CHANGES_REQUESTED** [review]
Please do not fix typos in any tests. I'm sorry if that wasn't clear, but we do not typically accept typo fixes for tests since they're really just churn and they may affect test output in the worst case. I very much do not want to see any unnecessary churn in test files unless a typo is gravely affecting the ability to understand the test, which I don't believe is true here.

As for the logistics, this should be *at least* split into different PRs for `compiler/`, `library/`, and all the tools within `src/*`.

As for the typos in the files under `compiler/`, which I would probably be personally able to review:

I would also appreciate if you did your own self-review of this again, and remove anything that you're not *certain* is a typo. For example, you changed `optin_builtin_traits` to `option_builtin_traits` -- I understand that this requires context, but this is *not* a typo. If you don't have the context, I suggest just leaving it alone. I would appreciate if you reverted most of the actual variable name changes. For example, variables are typically not typo'd, compared to comments which seem like they often are. An example is `inh_mut` (inh stands for inherited, not a typo of "in").

**Comment by fee1-dead** [comment]
> I would also appreciate if you did your own self-review of this again

You're referring to a different typo-fixing-person, I think.

**Comment by compiler-errors** [comment]
@fee1-dead: No, I'm referring to the "+ 2 hours of manual work removing false positives".

**Comment by fee1-dead** [comment]
I'd be happy to review this if the changes are only to `compiler/` and each PR is split to ~30 or more files each. Note that potential typos in `*.ftl` (like the `resolve_anonymous_livetime_non_gat_report_error`) need their corresponding identifiers in another file using it to also be changed `#[diag(resolve_anonymous_livetime_non_gat_report_error)]` or similar

**Comment by Sajjon** [comment]
> I'd be happy to review this if the changes are only to `compiler/` and each PR is split to ~40 or more files each. Note that potential typos in `*.ftl` (like the `resolve_anonymous_livetime_non_gat_report_error`) need their corresponding identifiers in the Rust file using it to also be changed `#[diag(resolve_anonymous_livetime_non_gat_report_error)]` or similar

Perfect, thank you very much! I've done this, and made an issue to track progress across the three PRs - https://github.com/rust-lang/rust/issues/129874. I'm closing this PR in favour of using that tracking issue.

**Pull request closed without merging**